tuxemon/npc.py

```diff
@@ -779,5 +883,52 @@ def alter_item_quantity(
 
         return success
 
+    def alter_item_quantity_and_money(
```

Comment:
this function is really taking on a lot of behavior.  it would be nice to see this split up into smaller parts so it can be easily tested and would make maintenance easier.